libcontainer/nsenter/nsexec.c

```diff
@@ -77,12 +78,18 @@ static int child_func(void *_arg)
     longjmp(*arg->env, 1);
 }
 
-static int clone_parent(jmp_buf *env, int flags) __attribute__((noinline));
-static int clone_parent(jmp_buf *env, int flags)
+static int clone_parent(jmp_buf *env, int flags, bool delay_ipc_unshare) __attribute__((noinline));
+static int clone_parent(jmp_buf * env, int flags, bool delay_ipc_unshare)
 {
 	struct clone_arg ca;
 	int		 child;
 
+	// Don't clone into NEWIPC at the same time as cloning into NEWUSER.
+	// This way we can ensure that NEWIPC namespace belongs to the root in new user namespace.
+	if (delay_ipc_unshare) {
```

Comment:
But as I said, the "proper" fix is to do `unshare(CLONE_NEWUSER)`, do all of the mapping and setgroup setup and then finally do the `clone`.